Classes/Toolbar/FLEXExplorerToolbar.m

```diff
@@ -150,9 +138,31 @@ - (void)layoutSubviews
     descriptionLabelFrame.size.width = CGRectGetMaxX(self.selectedViewDescriptionContainer.bounds) - kHorizontalPadding - descriptionOriginX;
     self.selectedViewDescriptionLabel.frame = descriptionLabelFrame;
 }
+    
+    
+#pragma mark - Setter Overrides
 
+- (void)setToolbarItems:(NSArray *)toolbarItems {
+    // Remove old toolbar items, if any
+    for (FLEXToolbarItem *item in _toolbarItems) {
+        [item removeFromSuperview];
+    }
+    
+    // Trim to 5 items if necessary
+    if (toolbarItems.count > 5) {
+        toolbarItems = [toolbarItems subarrayWithRange:NSMakeRange(0, 5)];
+    }
 
-#pragma mark - Setter Overrides
+    for (FLEXToolbarItem *item in toolbarItems) {
+        [self addSubview:item];
+    }
+
+    _toolbarItems = toolbarItems.copy;
+
+    // Lay out new items
+    [self setNeedsLayout];
+    [self layoutIfNeeded];
```

Comment:
I should clarify that the `setNeedsLayout` call is good and I only suggest removing the `layoutIfNeeded`. I poked into UITabBar out of curiosity, and didn't come across a `layoutIfNeeded` call in the surprisingly complicated path that comes from `setItems`.